src/nomad_measurements/xrd/nx.py

```diff
@@ -0,0 +1,184 @@
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_one/@units': 'archive.data.xrd_settings.source.kalpha_one.units',
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two': 'archive.data.xrd_settings.source.kalpha_two.magnitude',
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/k_alpha_two/@units': 'archive.data.xrd_settings.source.kalpha_two.units',
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/ratio_k_alphatwo_k_alphaone': 'archive.data.xrd_settings.source.ratio_kalphatwo_kalphaone',
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta': 'archive.data.xrd_settings.source.kbeta.magnitude',
+        '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/kbeta/@units': 'archive.data.xrd_settings.source.kbeta.units',
+    }
+
+    for key, archive_concept in concept_map.items():
+        if isinstance(archive_concept, dict):
+            if key == scan_type:
+                for sub_key, sub_archive_concept in archive_concept.items():
+                    _, arch_attr = sub_archive_concept.split('.', 1)
+                    value = None
+                    try:
+                        value = walk_through_object(archive, arch_attr)
+                    except (AttributeError, IndexError, KeyError, ValueError):
+                        pass
+                    finally:
+                        if value is not None:
+                            template[sub_key] = (
+                                str(value)
+                                if archive_concept.endswith('units')
+                                else value
+                            )
+            else:
+                continue
+        elif archive_concept:
+            _, arch_attr = archive_concept.split('.', 1)
+            value = None
+            try:
+                value = walk_through_object(archive, arch_attr)
+            # Use multiple excepts to avoid catching all exceptions
+            except (AttributeError, IndexError, KeyError, ValueError):
+                pass
+            finally:
+                if value is not None:
+                    template[key] = str(value) if key.endswith('units') else value
+
+    template['/ENTRY[entry]/definition'] = 'NXxrd_pan'
+
+    # Links to the data and concepts
+    template['/ENTRY[entry]/@default'] = 'experiment_result'
+    template['/ENTRY[entry]/experiment_result/@signal'] = 'intensity'
+    template['/ENTRY[entry]/experiment_result/@axes'] = 'two_theta'
+    template['/ENTRY[entry]/q_data/q'] = {
+        'link': '/ENTRY[entry]/experiment_result/q_norm'
+    }
+    template['/ENTRY[entry]/q_data/intensity'] = {
+        'link': '/ENTRY[entry]/experiment_result/intensity'
+    }
+    template['/ENTRY[entry]/q_data/q_parallel'] = {
+        'link': '/ENTRY[entry]/experiment_result/q_parallel'
+    }
+    template['/ENTRY[entry]/q_data/q_perpendicular'] = {
+        'link': '/ENTRY[entry]/experiment_result/q_perpendicular'
+    }
```

Comment:
I wonder why you are using "link" for only these quantities and not for others. What's the meaning of "link" in templates?

Reply:
This is how the `NXxrd_pan` app def has been defined which is similar to the reference, avoiding the duplication of array data.

Reply:
I see. Then I would suggest doing the same for other array quantities like two_theta, omega, etc. For now, we can stick with the current version.